nightfall-client/src/routes/commitment.mjs

```diff
@@ -88,6 +91,61 @@ router.get('/commitments', async (req, res, next) => {
   }
 });
 
+/**
+ * @description the endpoint that will send a reponse with all the
+ * existent commitments for the list of compressedPkd received in the
+ * request body.
+ * @author luizoamorim
+ */
+router.post('/allByCompressedZkpPublicKey', async (req, res, next) => {
+  logger.debug('commitment/allByCompressedZkpPublicKey endpoint received POST');
+  const arrayOfCompressedZkpPublicKey = req.body;
+  try {
+    const allCommitmentsByListOfCompressedZkpPublicKey =
+      await getAllCommitmentsByCompressedZkpPublicKeyList(arrayOfCompressedZkpPublicKey);
+    res.json({ allCommitmentsByListOfCompressedZkpPublicKey });
+  } catch (err) {
+    logger.error(err);
+    next(err);
+  }
+});
```

Comment:
I would move to the Rest approach in terms of nomenclature, like the following:
- Root path: `/commitments`
  - GET / (You already have this one!)
  - POST / (Instead of saveAll - having the swagger will give enough information this operation can save more than one object!)
  - etc.

This is a little guide: https://restfulapi.net/resource-naming/

What do you think?

Reply:
@israelboudoux so you suggest that instead of having 2 routes

a) `/` for get all
b) `/compressedPublicKeys` for get commitments filtering by compressedPub..

just have `a)`, right?

Two thoughts:

1. This part of the API already has a dedicated endpoint to filter by `salt` - so having `b)` wouldn't be "too alien" :alien:

2. Now we know that this compressedPub.. is not "sensitive", however I think we kept the POST method to avoid exceeding max query string length (although feels unlikely)

What do you say?? :blush:

Reply:
I'll not handle the root path now. Because we'll need to change all the calls. Let's do this refactor when we start to create the swagger documentation. 

I agree to remove the all so when we already have a plural.